xterm/shellserver.pl

```diff
@@ -116,7 +117,13 @@
 				exit(0);
 				}
 			if ($wsconn) {
-				$wsconn->send_utf8($buf);
+				# Do not print commands to display, if start from two
+				# consecutive spaces and ends with return char. This
+				# regex only catches commands sent in one shot using
+				# socket.send('  command\r') call, i.e. it won't work
+				# if command is pasted or typed in a normal way
+				$buf =~ s/^[\s]{2}\S+.*[\r\n]+[\e]\[\?2004l[\n]*//;
```

Comment:
This seems a bit hacky - I think we should just pass all output back, because this could trigger if something in the shell outputs a line that starts with two spaces.

Reply:
> because this could trigger if something in the shell outputs a line that starts with two spaces.

It should also match a new line and escape char, which is impossible.

Alright, I made it more resilient to impossible output - https://github.com/webmin/webmin/pull/1776/commits/163480b8591cdc4ccabccef4ff112d778afd1ced

Reply:
We could technically remove it, it's not critical but it would be nice to have a way to suppress output for the commands sent directly.

Reply:
Alright, I will drop it for now to move on. We can discuss it later.

As everything else is alright, I will merge it then.

Reply:
Thanks!